nativelib/src/main/resources/platform.c

```diff
@@ -33,3 +46,79 @@ int scalanative_little_endian() {
     int n = 1;
     return (*(char *)&n);
 }
+
+#ifndef __APPLE__
+void scalanative_set_os_props(void (*add_prop)(const char *, const char *)) {
+#ifdef _WIN32
```

Comment:
Note that the windows code compiles, but I could not wire things up on my vm to actually get scala native to run on windows.

Reply:
If we can't test it, I'm afraid it's a dead and unmaintained code that might or might not work in the future. I think we should cross that bridge when we get there, and not prematurely.

Reply:
FWIW, I did add the following to platform.c
```c
static void print_prop(const char *name, const char *value) {
  printf("Added prop %s %s\n", name, value);
}
int main(int argc, char **argv) {
  scalanative_set_os_props(&print_prop)
}
```

And it printed:
```
Added os.name Windows 8
Added os.version 6.2
Added java.io.tmpdir C:\Users\ethan\AppData\Local\Temp
```

The os is actually wrong because of changes made to GetVersionEx: https://docs.microsoft.com/en-us/windows/desktop/SysInfo/targeting-your-application-at-windows-8-1. To fix, there would have to be a build step that adds a manifest to the generated binary. But at least the os.name prefix is correct and it gets the right temp directory.